backend/tests/task/test_api.py

```diff
@@ -1,17 +1,71 @@
+import json
+
+from grant.utils import totp_2fa
+from grant.task.jobs import MilestoneDeadline
 from datetime import datetime, timedelta
 
 from grant.task.models import Task, db
 from grant.task.jobs import PruneDraft
 from grant.milestone.models import Milestone
-from grant.proposal.models import Proposal
-from grant.utils.enums import ProposalStatus, Category
+from grant.proposal.models import Proposal, ProposalUpdate
+from grant.utils.enums import ProposalStatus, ProposalStage, Category
+
+from ..config import BaseProposalCreatorConfig
+from ..test_data import mock_blockchain_api_requests
 
 from mock import patch, Mock
 
-from ..config import BaseProposalCreatorConfig
+test_update = {
+    "title": "Update Title",
+    "content": "Update content."
+}
 
+milestones_data = [
+    {
+        "title": "All the money straightaway",
+        "content": "cool stuff with it",
+        "days_estimated": 30,
+        "payout_percent": "100",
+        "immediate_payout": False
+    }
+]
 
 class TestTaskAPI(BaseProposalCreatorConfig):
+    def p(self, path, data):
+        return self.app.post(path, data=json.dumps(data), content_type="application/json")
+
```

Comment:
I swiped `login_admin` from the admin tests since I needed admin functionality, but there's likely a more DRY way to implement this.